### PR TITLE
Reacting to events

### PR DESCRIPTION
**Pull request by jamesmorgan**
* [x] React to ERC721 token movement events to update `owner`
* [x] React to Card attribute changes to rebuild data
* [x] Refactor duplication and clean up switch statement
* [x] Split `card` and `event` store service up
	* [x] Refactor firesotre into logic to prevent error
* [x] Add API to query for a users best team
	* [x] load players in `position` for `address`
	* [x] expose public API to get top `squad`

### CODE REVIEW

File: functions/api/squad/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        };
+
+        const keeper = await carrdService.getTopPlayerInPositionForAddress(network, address, GOAL_KEEPER, formation.keeper);
+
+        const defence = await carrdService.getTopPlayerInPositionForAddress(network, address, DEFENDER, formation.defence);
+
+        const midfield = await carrdService.getTopPlayerInPositionForAddress(network, address, MIDFIELDER, formation.midfield);
+
+        const strikers = await carrdService.getTopPlayerInPositionForAddress(network, address, STRIKER, formation.strikers);
+
```
Comment (andygray) [inline functions/api/squad/index.js]: `getTopPlayersInPositionForAddress` players instead of player?

File: functions/api/squad/index.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                squadTotal: squadTotal,
+                squadAverage: squadAverage,
+                team: {
+                    keeper: _.size(keeper) === 1 ? keeper[0] : {},
+                    defence: defence,
+                    midfield: midfield,
+                    strikers: strikers
+                }
+            });
+    } catch (e) {
```
Comment (andygray) [inline functions/api/squad/index.js]: 👍

File: functions/index.js [hunk 3]
```diff
@@ -53,32 +61,46 @@ exports.newEventTrigger =
...
-                            network,
-                            tokenId,
-                        );
-                        await require('./services/data/eventsStore.service').upsertAttrsAvg(
-                            network,
-                            tokenId,
-                            tokenDetails.attributeAvg,
-                        );
-                        break;
+            const event = get(document, 'event');
```
Comment (andygray) [inline functions/index.js]: document can be null? guessing the `get` is null safe

File: functions/api/squad/index.js (hunk 1 above)
Comment (jamesmorgan) [reply functions/api/squad/index.js]: 👍

File: functions/index.js (hunk 3 above)
Comment (jamesmorgan) [reply functions/index.js]: yes, null safe

**Pull request merged**